nanoFramework.System.Net/NetworkHelper/NetworkHelper.cs

```diff
@@ -0,0 +1,300 @@
+        {
+            _requiresDateTime = requiresDateTime;
+
+            InternalSetup(true);
+
+            // fire working thread
+            new Thread(WorkingThread).Start();
+        }
+
+        /// <summary>
+        /// This method will setup the network configurations so that the <see cref="NetworkReady"/> event will fire when the required conditions are met.
+        /// That will be the network connection to be up, having a valid IpAddress and optionally for a valid date and time to become available.
+        /// </summary>
+        /// <param name="ipConfiguration">The static IP configuration you want to apply.</param>
+        /// <param name="requiresDateTime">Set to <see langword="true"/> if valid date and time are required.</param>
+        /// <exception cref="NotSupportedException">There is no network interface configured. Open the 'Edit Network Configuration' in Device Explorer and configure one.</exception>
+        public static void SetupNetworkHelper(
+            IPConfiguration ipConfiguration,
+            bool requiresDateTime = false)
+        {
+            _requiresDateTime = requiresDateTime;
+            _ipConfiguration = ipConfiguration;
+
+            InternalSetup(true);
+
+            // fire working thread
+            new Thread(WorkingThread).Start();
+        }
+
+        /// <summary>
+        /// This will wait for the network connection to be up and optionally for a valid date and time to become available.
+        /// </summary>
+        /// <param name="token">A <see cref="CancellationToken"/> used for timing out the operation.</param>
+        /// <param name="requiresDateTime">Set to <see langword="true"/> if valid date and time are required.</param>
+        /// <returns><see langword="true"/> on success. On failure returns <see langword="false"/> and details with the cause of the failure are made available in the ConnectionError property</returns>
+        public static bool SetupAndConnectNetwork(
+            CancellationToken token = default,
+            bool requiresDateTime = false)
+        {
+            return SetupAndConnectNetwork(null, token, requiresDateTime);
+        }
+
+        /// <summary>
+        /// This will wait for the network connection to be up and optionally for a valid date and time to become available.
+        /// </summary>
+        /// <param name="ipConfiguration">The static IPv4 configuration to apply to the Ethernet network interface.</param>
+        /// <param name="token">A <see cref="CancellationToken"/> used for timing out the operation.</param>
+        /// <param name="requiresDateTime">Set to <see langword="true"/> if valid date and time are required.</param>
+        /// <returns><see langword="true"/> on success. On failure returns <see langword="false"/> and details with the cause of the failure are made available in the ConnectionError property</returns>
+        public static bool SetupAndConnectNetwork(
```

Comment:
Something missing in all constructors is the interface to setup and check. By default there are 3. So I may want to setup my wifi one or my ethernet one. And in the current logic, if I have any, if will just be ok. And that's not what I can expect.

Reply:
Here it "defaults" to Ethernet. When used from the wifi one it will chnage the filed above to the appropriate interface type.

Reply:
We still should have a method to check the one that you want to check. In a mixed scenario with wifi and ethernet it's really needed